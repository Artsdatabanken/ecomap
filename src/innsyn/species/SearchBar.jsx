import React from 'react'
import IconMenu from 'material-ui/IconMenu'
import IconButton from 'material-ui/IconButton'
import FontIcon from 'material-ui/FontIcon'
import NavigationExpandMoreIcon from 'material-ui/svg-icons/navigation/expand-more'
import ActionSearch from 'material-ui/svg-icons/action/search'
import MenuItem from 'material-ui/MenuItem'
import DropDownMenu from 'material-ui/DropDownMenu'
import RaisedButton from 'material-ui/RaisedButton'
import AutoComplete from 'material-ui/AutoComplete'
import {
  Toolbar,
  ToolbarGroup,
  ToolbarSeparator,
  ToolbarTitle
} from 'material-ui/Toolbar'

export default class ToolbarExamplesSimple extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      value: 3
    }
  }

  handleChange = (event, index, value) => this.setState({ value })

  render () {
    return (
      <Toolbar>
        <ToolbarGroup firstChild>
          <ActionSearch />
          <AutoComplete
            style={{paddingLeft: '100px', width: '40%'}}
            hintText='SÃ¸k...'
            filter={AutoComplete.fuzzyFilter}
            dataSource={[]}
            maxSearchResults={15}
            underlineShow={false}
            onUpdateInput={v => this.props.onChange(v)}
          />
        </ToolbarGroup>
      </Toolbar>
    )
  }
}
