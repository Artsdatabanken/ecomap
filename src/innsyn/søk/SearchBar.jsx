import PropTypes from 'prop-types'
import React, { Component } from 'react'
import IconButton from 'material-ui/IconButton'
import NavigationMenu from 'material-ui/svg-icons/navigation/menu'
import Paper from 'material-ui/Paper'
import ActionSearch from 'material-ui/svg-icons/action/search'

import MainSearchBox from './MainSearchBox'

class SearchBar extends Component {
  static contextTypes = {
    muiTheme: PropTypes.object.isRequired
  };

  static propTypes = {
    onLeftIconButtonTouchTap: PropTypes.func,
    onRightIconButtonTouchTap: PropTypes.func,
    onUpdateSearch: PropTypes.func.isRequired,
    zDepth: PropTypes.number
  };

  static defaultProps = {
    onLeftIconButtonTouchTap: undefined,
    onRightIconButtonTouchTap: undefined,
    zDepth: 2
  };

  handleTouchTapLeftIconButton = (event) => {
    if (this.props.onLeftIconButtonTouchTap) {
      this.props.onLeftIconButtonTouchTap(event)
    }
  };

  handleTouchTapRightIconButton = (event) => {
    if (this.props.onRightIconButtonTouchTap) {
      this.props.onRightIconButtonTouchTap(event)
    }
  };

  render () {
    const {
      onUpdateSearch,
      zDepth
    } = this.props

    const styles = {
      root: {
        position: 'relative',
        zIndex: this.context.muiTheme.zIndex.appBar,
        width: '100%',
        display: 'flex',
        paddingLeft: 0,
        paddingRight: 0
      }
    }

    const menuElementLeft = (
      <IconButton
        style={{ marginLeft: '4px', marginRight: '14px' }}
        onTouchTap={this.handleTouchTapLeftIconButton}
      >
        <NavigationMenu style={Object.assign({}, styles.iconButtonIconStyle)} />
      </IconButton>
    )

    const menuElementRight = (
      <IconButton
        style={{ marginRight: 0, marginLeft: 'auto' }}
        onTouchTap={this.handleTouchTapRightIconButton}
      >
        <ActionSearch color='#aaaaaa' hoverColor='#333333' />
      </IconButton>
    )

    return (
      <Paper
        rounded
        style={styles.root}
        zDepth={zDepth}
      >
        {menuElementLeft}
        <MainSearchBox onUpdateSearch={onUpdateSearch} />
        {menuElementRight}
      </Paper>
    )
  }
}

export default SearchBar
