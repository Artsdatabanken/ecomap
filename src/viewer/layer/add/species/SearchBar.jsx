import React from 'react'
import ActionSearch from 'material-ui/svg-icons/action/search'
import AutoComplete from 'material-ui/AutoComplete'

export default class ToolbarExamplesSimple extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      value: 3
    }
  }
  componentDidMount () {
    this.searchInput.focus()
  }
  handleChange = (event, index, value) => this.setState({ value })

  render () {
    return (
      <span>
        <ActionSearch style={{ position: 'relative', left: 16, top: 8 }} />
        <AutoComplete
          style={{ left: 32, width: '40%' }}
          hintText='SÃ¸k...'
          filter={AutoComplete.fuzzyFilter}
          dataSource={[]}
          maxSearchResults={15}
          underlineShow={false}
          ref={(input) => { this.searchInput = input }}
          onUpdateInput={v => this.props.onChange(v)}
        />
      </span>
    )
  }
}
