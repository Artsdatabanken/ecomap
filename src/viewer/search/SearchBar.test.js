import injectTapEventPlugin from 'react-tap-event-plugin'
import React from 'react'
import ReactDOM from 'react-dom'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

import SearchBar from './SearchBar'

// Needed for onTouchTap
// http://stackoverflow.com/a/34015469/988941
injectTapEventPlugin()

it('renders ok', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <MuiThemeProvider>
      <SearchBar onUpdateSearch={() => {}} />
    </MuiThemeProvider>
, div)
})
