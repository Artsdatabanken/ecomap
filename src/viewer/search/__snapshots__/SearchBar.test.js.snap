// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ok 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "paddingLeft": 0,
      "paddingRight": 0,
      "position": "relative",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1100,
    }
  }
>
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "border": 10,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 0,
        "fontWeight": "inherit",
        "height": 48,
        "margin": 0,
        "marginLeft": "4px",
        "marginRight": "14px",
        "muiPrepared": true,
        "outline": "none",
        "overflow": "visible",
        "padding": 12,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": 48,
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "currentColor",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </div>
  </button>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "muiPrepared": true,
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <div
        className={undefined}
        style={
          Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 48,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "bottom": 12,
              "color": "#aaaaaa",
              "muiPrepared": true,
              "opacity": 1,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          SÃ¸k i Naturtyper
        </div>
        <input
          autoComplete="off"
          disabled={false}
          id="mainsearchinput"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          open={false}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="text"
          value=""
        />
      </div>
      <div
        style={
          Object {
            "display": "none",
          }
        }
      />
    </div>
  </div>
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "border": 10,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 0,
        "fontWeight": "inherit",
        "height": 48,
        "margin": 0,
        "marginLeft": "auto",
        "marginRight": 0,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "visible",
        "padding": 12,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": 48,
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "#aaaaaa",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
  </button>
</div>
`;
